editors/code/src/commands/inlay_hints.ts

```diff
@@ -0,0 +1,100 @@
+import * as vscode from 'vscode';
+import {
+    Range,
+    TextDocumentChangeEvent,
+    TextEditor
+} from 'vscode';
+import { TextDocumentIdentifier } from 'vscode-languageclient';
+import { Server } from '../server';
+
+interface InlayHintsParams {
+    textDocument: TextDocumentIdentifier;
+}
+
+interface InlayHint {
+    range: Range;
+    kind: string;
+    label: string;
+}
+
+const typeHintDecorationType = vscode.window.createTextEditorDecorationType({
+    after: {
+        color: new vscode.ThemeColor('ralsp.inlayHint')
+    }
+});
+
+export class HintsUpdater {
+    private displayHints = true;
+
+    public async loadHints(
+        editor: vscode.TextEditor | undefined
+    ): Promise<void> {
+        if (this.displayHints && editor !== undefined) {
```

Comment:
We should check if `editor` is displaying a Rust file. I see that this is done in `updateHints`, but not here. Perhaps a common update logic can be factored out?

Reply:
Thanks, adjusted.